crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -305,32 +311,98 @@ impl ConfigManagerActor {
         &mut self,
         topic: &str,
         result: DownloadResult,
-    ) -> Result<(), ChannelError> {
-        if let Some(ConfigOperation::Update(mut request)) = self.pending_operations.remove(topic) {
-            let topic = Topic::new_unchecked(topic);
-            match result {
-                Ok(response) => {
-                    request.successful(response.file_path.as_path().to_str().unwrap());
-                    info!(
-                        "Config Update request processed for config type: {}.",
-                        request.config_type
-                    );
-                    self.publish_command_status(&topic, &ConfigOperation::Update(request))
-                        .await?;
-                }
-                Err(err) => {
-                    let error_message =
-                        format!("tedge-configuration-plugin failed downloading a file: {err}",);
-                    request.failed(&error_message);
-                    error!("{}", error_message);
-                    self.publish_command_status(&topic, &ConfigOperation::Update(request))
-                        .await?;
-                }
+    ) -> Result<(), ConfigManagementError> {
+        let Some(ConfigOperation::Update(mut request)) = self.pending_operations.remove(topic)
+        else {
+            return Ok(());
+        };
+
+        let topic = Topic::new_unchecked(topic);
+
+        let response = match result {
+            Ok(response) => response,
+            Err(err) => {
+                let error_message =
+                    format!("tedge-configuration-plugin failed downloading a file: {err}",);
+                request.failed(&error_message);
+                error!("{}", error_message);
+                self.publish_command_status(&topic, &ConfigOperation::Update(request))
+                    .await?;
+                return Ok(());
             }
-        }
+        };
+
+        // new config was downloaded into tmpdir, we need to write it into destination using tedge-write
+        let from = Utf8Path::from_path(response.file_path.as_path()).unwrap();
+
+        let deployed_to_path = match self.deploy_config_file(from, &request.config_type) {
+            Ok(path) => path,
+            Err(err) => {
+                let error_message = format!(
+                    "Handling of operation failed with: Could not write to config file: {err}",
```

Comment:
```suggestion
                    "config-manager failed writing updated configuration file: {err}",
```

(I am changing the one above to say `config-manager` instead of `tedge-configuration-plugin` as the latter is too specific in the case of `tedge-agent-v1`. That case is checked by a robot test, so I'm dealing with that in #2493, but it would be good to avoid adding new ones).